tests/trainer/test_trainer.py

```diff
@@ -40,6 +40,24 @@ def config(self, rank_zero_seed: int):
             'seed': rank_zero_seed,
         }
 
+    def test_init_errors_when_using_fp16_and_not_deepspeed(self, config):
+        config['precision'] = Precision.FP16
+
+        with pytest.raises(ValueError):
+            Trainer(**config)
+
+    @pytest.mark.gpu
+    @pytest.mark.parametrize("precision", [Precision.AMP, Precision.FP16, Precision.FP32])
+    def test_trainer_with_deepspeed(self, config, precision: Precision):
```

Comment:
I know this technically isn't a trainer init test, but this will be refactored in #942, which completely refactors the trainer tests.